integrations/weaviate/src/haystack_integrations/document_stores/weaviate/document_store.py

```diff
@@ -238,72 +233,42 @@ def _to_document(self, data: Dict[str, Any]) -> Document:
 
         return Document.from_dict(data)
 
-    def _query_paginated(self, properties: List[str], cursor=None):
-        collection_name = self._collection_settings["class"]
-        query = (
-            self._client.query.get(
-                collection_name,
-                properties,
+    def _query_paginated(self, properties: List[str]):
+
+        try:
+            result = self._collection.iterator(include_vector=True, return_properties=properties)
+        except weaviate.exceptions.WeaviateQueryError as e:
+            msg = f"Failed to query documents in Weaviate. Error: {e.message}"
+            raise DocumentStoreError(msg) from None
+        return list(result)
+
+    def _query_with_filters(self, filters: weaviate.collections.classes.filters.Filter) -> List[Dict[str, Any]]:
+
+        try:
+            # this is the default value for max number of objects to retrieve in Weaviate
+            # see QUERY_MAXIMUM_RESULTS
+            # see https://weaviate.io/developers/weaviate/config-refs/env-vars#overview
+            # and https://weaviate.io/developers/weaviate/api/graphql/additional-operators#pagination-with-offset
+            limit = 10_000
+            result = self._collection.query.fetch_objects(
+                filters=convert_filters(filters), include_vector=True, limit=limit
             )
-            .with_additional(["id vector"])
-            .with_limit(100)
-        )
+        except weaviate.exceptions.WeaviateQueryError as e:
+            msg = f"Failed to query documents in Weaviate. Error: {e.message}"
+            raise DocumentStoreError(msg) from None
```

Comment:
Shouldn't we reraise?
```suggestion
            raise DocumentStoreError(msg) from e
```

Reply:
I felt the traceback from WeaviateQueryError is an implementation detail that the user does not need to know, since the problem would already be clear from `msg`

Reply:
I would leave it, I think it's useful in any case.